File: frontend-angular/src/app/models/Proyecto.ts
export class Proyecto {
    private _id: number = -1;
    private _titulo: string = '';
    private _descripcion: string = '';
    private _url: string = '';
    private _imagenUrl: string = '';

    public constructor() {}

    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

    public get titulo(): string {
        return this._titulo;
    }
    public set titulo(value: string) {
        this._titulo = value;
    }

    public get descripcion(): string {
        return this._descripcion;
    }
    public set descripcion(value: string) {
        this._descripcion = value;
    }

    public get url(): string {
        return this._url;
    }
    public set url(value: string) {
        this._url = value;
    }

    public get imagenUrl(): string {
        return this._imagenUrl;
    }
    public set imagenUrl(value: string) {
        this._imagenUrl = value;
    }
}

File: backend-spring-boot/src/main/java/com/facundocastro/api/model/Proyecto.java
package com.facundocastro.api.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.sun.istack.NotNull;
import lombok.Data;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;

@Data
@Entity
public class Proyecto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    private String titulo;
    @NotNull
    @Lob
    private String descripcion;
    private String url;
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JsonIgnore
    private Usuario usuario;
}
